content/chapters/compute/lab/README.md

```diff
@@ -282,6 +282,201 @@ Therefore, there is also a gray area where there is no clear right answer to thi
+root@56b9a761d598:/usr/local/apache2# ps -ef
+UID          PID    PPID  C STIME TTY          TIME CMD
+root           1       0  0 20:38 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       9       1  0 20:38 pts/0    00:00:00 httpd -DFOREGROUND
+www-data      10       1  0 20:38 pts/0    00:00:00 httpd -DFOREGROUND
+root          25       0  0 20:40 pts/1    00:00:00 bash
+root          31      25  0 20:40 pts/1    00:00:00 ps -ef
+```
+
+We see 3 `httpd` processes. The first one, running as root, is the main process, while the other 2 are the workers.
+
+Let's confirm that we are using the `event` mpm:
+
+```Bash
+root@56b9a761d598:/usr/local/apache2# grep mod_mpm conf/httpd.conf
+LoadModule mpm_event_module modules/mod_mpm_event.so
+#LoadModule mpm_prefork_module modules/mod_mpm_prefork.so
+#LoadModule mpm_worker_module modules/mod_mpm_worker.so
+```
+
+The `event` mpm is enabled, so we expect each worker to be multi-threaded. Let's check:
+
+```Bash
+root@56b9a761d598:/usr/local/apache2# ps -efL
+UID          PID    PPID     LWP  C NLWP STIME TTY          TIME CMD
+root           1       0       1  0    1 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       8       1       8  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       8       1      11  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       8       1      12  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       8       1      16  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       8       1      17  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       8       1      18  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       8       1      19  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       9       1       9  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       9       1      14  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       9       1      15  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       9       1      20  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       9       1      21  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       9       1      22  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+www-data       9       1      23  0    7 20:56 pts/0    00:00:00 httpd -DFOREGROUND
+root          24       0      24  1    1 20:56 pts/1    00:00:00 bash
+root          30      24      30  0    1 20:56 pts/1    00:00:00 ps -efL
+```
+
+Indeed, each worker has 7 threads. In fact, the number of threads per worker is configurable, as well as the number of initial workers.
+
+When a new connection arrives, it will be handled by whatever thread is available from any worker. If all the threads are busy, then the server will spawn more worker processes (and therefore more threads), as long as the total number of threads is below some threshold, which is also configurable.
```

Comment:
```suggestion
When a new connection is created, it will be handled by whatever thread is available from any worker.
If all threads are busy, then the server will spawn more worker processes (and therefore more threads), as long as the total number of threads is below some threshold, which is also configurable.
```